src/main/java/software/amazon/cloudformation/LambdaWrapper.java

```diff
@@ -471,6 +477,21 @@ private void publishExceptionMetric(final Action action, final Throwable ex, fin
         }
     }
 
+    /*
+     * null-safe exception metrics delivery
+     */
+    private void
+        publishExceptionCodeAndCountMetric(final Action action, final HandlerErrorCode handlerErrorCode, final boolean thrown) {
+        if (this.metricsPublisherProxy != null) {
+            EnumSet.allOf(HandlerErrorCode.class).stream()
+                // publishing 0 value for all (if not thrown) otherwise filtered
+                .filter(errorCode -> errorCode.equals(handlerErrorCode) || !thrown)
```

Comment:
This line can be removed.  We actually want to do forEach for all ErrorCode, not match in what cases.

In case of in a Failed cases, this will only emit metric 1 for the one equals to the ErrorCode, but won't emit 0 metric for other metrics.